### ISSUE TITLE
[Feature Request] Bulk delete findings after importing findings (Nessus) instead of clicking through each individually

### ISSUE THREAD

**Issue by Gray-0men** [Syslifters/sysreptor#218]
Hoping for a bulk delete/multi select like option after parsing in Nessus data, like in this scenario I would just delete all of these arguably repeated ones and just use the most recent vuln and summarize it. But at the moment I would have to click on each finding and select delete at the top right with a confirmation, just a little tedious to remove.

![image](https://github.com/Syslifters/sysreptor/assets/39830524/47085813-d2f2-42f5-a400-24262e24b54c)

**Comment by aronmolnar** [comment]
We can do this :)

**Comment by syn-4ck** [comment]
Hi @aronmolnar & @Gray-0men, This can be implemented generally (any finding template), instead of only Nessus one? :)

**Comment by aronmolnar** [comment]
Yes, we do not (cannot) tag findings imported by nessus anyway, so this needs to be a generic solution.

I think the best option is to define a search query for finding title and to delete all matched findings (and of course provide the option of a dry run)

**Comment by MWedl** [comment]
A more user friendly solution would be to select multiple findings in the sidebar and bulk delete them. Selection could be implemented similar to file browsers or directory trees in IDEs (e.g. VSCode file explorer). 

I think selecting findings via queries is not user friendly, especially when you want to delete findings with unrelated titles. This feels more like working with a database than an UI application.

**Comment by Gray-0men** [comment]
yeah just noticing the types of findings Nessus sometimes finds and it would be nice to be able to bulk select findings and delete, so just a generic solution. Was just using Nessus to get those results so that's why I mentioned Nessus.

since importing findings from Nessus is an option this would just dramatically help out my workflow.

**Comment by StephanPillhofer** [comment]
> > Yes, we do not (cannot) tag findings imported by nessus anyway, so this needs to be a generic solution.
> > I think the best option is to define a search query for finding title and to delete all matched findings (and of course provide the option of a dry run)
> 
> Also that would be fantastic, if there was a way I could do a findings search like "intitle:Adobe Flash Player" or maybe even "host:10.10.10.10" and see a filtered list of all those items I can then checkbox select the ones or 'select all' and delete that would be amazing

I like the idea but then you would need to know the extact name of the field you are filtering for. On the other side, if you create a UI component for every possible field then it could get really crowded like in the filter component of DefectDojo which I really don't like because it's super crowded ...

**Comment by aronmolnar** [comment]
We discussed internally and we (at least Michael and I) obviously had a misunderstanding of where to implement the bulk delete features: in reptor vs SysReptor UI.

What I meant was that we can implement the feature in reptor.  
Michael meant that selecting findings via queries wouldn't be nice via the web UI.

For the short term, reptor is much easier to implement and I just released it in v0.12. (`pip3 install --upgrade reptor`)

```bash
reptor deletefindings --title-contains "delete me"  # Delete findings matching the search query
reptor deletefindings --exclude-title-contains "leave me"  # Exclude findings with search query
reptor deletefindings --no-dry-run  # Delete all findings, no dry run
```

(For sake of simplicity, matching works for titles only, at the moment.)

For the long term, it would be nice to allow multi-selecting findings (e.g., by pressing CTRL to select individual findings or Shift to select ranges, like we're used to in file explorers).  
This will, however, require more efforts and is not on our roadmap currently.
